### PR TITLE
Fix #461

### LINKED ISSUE #461: When retranslating the UI, the settings menu is partially translated
This seems a QT bug, but it must be investigated a bit more. The UI is fully re-translated when the user changes the language, except for the settings view.

### PR DESCRIPTION
**Pull request by lesleyjanenorton**
Changes the way we build the settings options so that they are able to respond to language changes. Fixes #461

### CODE REVIEW

**Review by bakulf — APPROVED** [review]

**Comment by bakulf** [comment]
Thanks! do you want to land it in the next android release?

**Comment by lesleyjanenorton** [comment]
Your call! It's a big(ish) restructure so I'm fine to wait until the next release if that feels safer.

**Comment by bakulf** [comment]
Can you rebase the patch? I would like to merge it.

**Pull request merged**
